### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by mcastorena0316**
Hi, i think im ready for my first review. Thanks!

### CODE REVIEW

**Review by Ispirett — CHANGES_REQUESTED** [review]
### Status Changes Required
@mcastorena0316  great work so far :100:  :+1: 
There are  a lot of errors when testing your code :bug: 

##### Errors
- [ ] 
  **ArgumentError:
       wrong number of arguments (given 0, expected 1)**

My_each, my_each_with index, my_select, my_all, my_any, my_none my_count, and my_inject Should all have parameters

- [ ]  ```LocalJumpError:
        no block given (yield)```
My_inject!  You need a guard case, at the top of your function,    return to_enum unless block_given?


**Suggestion**
You  can test your methods against the orignal methods example: array.my_none?(2) == array.none?(2)

**Comment by mcastorena0316** [comment]
hi, I follow your suggestions and test my methods against the originals and its currently working. Also I add the guardcase as recommended. Thanks!

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Hey good job @mcastorena0316 
You have done well in implementing the features for this project. I tested your methods and found they fail against some edge cases. We need to fix those failing cases before we can merge this PR. 
Please look at my comments and make changes accordingly.

### Martin Afani

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+  def my_all?
+    i = 0
+    while i < size
+      return false if yield(self[i]) == false || yield(self[i]).nil?
+
+      i += 1
+    end
+    true
+  end
```
Comment (whiz25) [inline enumerable.rb]: - [ ] when no block or argument is given returns true when none of the collection members are false or nil
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    contador = 0
+    i = 0
+    while i < size
+      contador += 1 if yield(self[i]) == true
+      i += 1
+    end
+    return false if contador.zero?
+
+    true
+  end
```
Comment (whiz25) [inline enumerable.rb]: - [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] false if none of the collection matches the pattern

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+  def my_none?
+    i = 0
+    while i < size
+      return false if yield(self[i]) == true
+
+      i += 1
+    end
+    true
+  end
```
Comment (whiz25) [inline enumerable.rb]: - [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  def my_count
+    contador = 0
+    i = 0
+    while i < size
+      contador += 1 if yield(self[i])
+
+      i += 1
+    end
+    contador
+  end
```
Comment (whiz25) [inline enumerable.rb]: - [ ] returns the number of items in enum through enumeration
- [ ] counts the number of items in enum that are equal to item if an argument is given

File: enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+      new_array4 << if block_given?
+                      yield(self[i])
+                    else
+                      mycall_proc.call(self[i])
+                    end
+      i += 1
+    end
+    new_array4
+  end
+end
```
Comment (whiz25) [inline enumerable.rb]: - [ ] returns an Enumerator if no block is given

File: enumerable.rb [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    return to_enum unless block_given?
+
+    result = start
+    i = 0
+    while i < size
+      result = yield(result, (self[i]))
+      i += 1
+    end
+    result
+  end
```
Comment (whiz25) [inline enumerable.rb]: - [x] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method

**Comment by mcastorena0316** [comment]
Hi, I did the reccommended changes and try it a bunch of times with different scenarios and variables, lets hope this time its the winner. Thanks!

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

Hi mcastorena0316

The corrections pointed out  by **```whiz25```** hasn't been implemented correctly; there are still failing the test cases.

Please use the examples in this [document](https://ruby-doc.org/core-2.6.5/Enumerable.html) as a guide while testing out the methods. 

After implementing those changes submit another form for a code review.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

**Comment by mcastorena0316** [comment]
Thanks for your review, i found my mistakes and i correct them.

File: enumerable.rb [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    x = args[0].nil? || args[0].class == Symbol ? 0 : args[0]
+    sym = args[0].class == Symbol ? args[0] : args[1].class == Symbol ? args[1] : nil
+    my_each do |elem|
+      x = if sym
+            x.send(sym, elem)
+          else
+            yield(x, elem)
+          end
+    end
+    x
```
Comment (wathigo) [inline enumerable.rb]: **[REQUIRED]**
* combines all elements of enum by applying a binary operation, specified by a block.
```
Failure/Error: let(:search) { proc { |memo, word| memo.length > word.length ? memo : word } }
     
     NoMethodError:
       undefined method `length' for 0:Integer
```

**Review by wathigo — CHANGES_REQUESTED** [review]
# Change Required  
## Great progress team :clap:  
We only have one issue to address, See the comment :point_down:  
### After implementing the change, submit another request for review.

**Happy Coding :computer: 
Simon Wathigo**

**Comment by mcastorena0316** [comment]
Thanks for explaining me the issue,  I believe I fixed it this error and doesn't appear anymore.

:four_leaf_clover: this is a lucky charm for luck to approve my project

**Review by imhta — CHANGES_REQUESTED** [review]
# Changes Required 🛑 

### Hello @mcastorena0316 , 👋 You have done a good job with this project but still there is a test failling in your `my_inject` method.

see comments for more details

----
### Don't forgot to request for another code review
----

Happy coding,
** Pon Muthu Selvam N**
_If you have any queries, feel free to ping me on slack_

File: enumerable.rb [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+      elsif !args[1].nil?
+        sym = args[1]
+        result = args[0]
+        new_array.my_each do |x|
+          result = result.send(sym, x)
+        end
+      end
+    end
+    result
+  end
```
Comment (imhta) [inline enumerable.rb]: Your `my_inject` method is failing a test
```bash
  1) enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
```

**Comment by mcastorena0316** [comment]
Thanks, changes made

**Review by wathigo — APPROVED** [review]
# Approved :smile:  
## Great work implementing all the requested changes :+1:  

### You  may move on :next_track_button:  

**Happy Coding :computer: 
Simon Wathigo**

**Pull request merged**